extensions/blocks/opentable/edit.js

```diff
@@ -0,0 +1,294 @@
+		setAttributes( validatedNewAttributes );
+	};
+
+	const styleOptions = getStyleOptions( rid );
+	const styleValues = getStyleValues( rid );
+
+	const updateStyle = newStyle => {
+		setAttributes( { style: newStyle } );
+	};
+
+	const getTypeAndTheme = fromStyle =>
+		rid.length > 1
+			? [ 'multi', 'button' !== fromStyle ? fromStyle : 'standard' ]
+			: [
+					'button' === fromStyle ? 'button' : 'standard',
+					'button' === fromStyle ? 'standard' : fromStyle,
+			  ];
+
+	const blockPreview = styleOveride => {
+		const [ type, theme ] = getTypeAndTheme( styleOveride ? styleOveride : style );
+		return (
+			<>
+				<div className={ `${ className }-overlay` }></div>
+				<iframe
+					title={ `Open Table Preview ${ clientId }` }
+					src={ `https://www.opentable.com/widget/reservation/canvas?rid=${ join(
+						rid,
+						'%2C'
+					) }&type=${ type }&theme=${ theme }&overlay=false&domain=${ domain }&lang=${
+						lang && languageValues.includes( lang ) ? lang : 'en-US'
+					}&newtab=${ newtab }&disablega=true` }
+				/>
+			</>
+		);
+	};
+
+	const blockControls = (
+		<BlockControls>
+			{ ! isEmpty( rid ) && (
+				<Toolbar
+					isCollapsed={ true }
+					icon="admin-appearance"
+					label={ __( 'Type', 'jetpck' ) }
+					controls={ styleOptions.map( styleOption => ( {
+						title: styleOption.label,
+						isActive: styleOption.value === style,
+						onClick: () => updateStyle( styleOption.value ),
+					} ) ) }
+				/>
+			) }
+		</BlockControls>
+	);
+
+	const onPickerSubmit = input => {
+		if ( Array.isArray( input ) ) {
+			setAttributes( {
+				rid: input,
+				style: input.length > 1 && 'button' === style ? 'standard' : style,
+			} );
+		} else {
+			parseEmbedCode( input );
+		}
+	};
+
+	const inspectorControls = () => (
+		<InspectorControls>
+			<PanelBody title={ __( 'Styles', 'jetpack' ) }>
+				<div className="block-editor-block-styles">
+					{ styleOptions.map( styleOption => {
+						return (
+							<div
+								key={ styleOption.value }
+								className={ classnames( 'block-editor-block-styles__item is-opentable', {
+									'is-active': styleOption.value === style,
+								} ) }
+								onClick={ () => updateStyle( styleOption.value ) }
+								onKeyDown={ event => {
+									if ( ENTER === event.keyCode || SPACE === event.keyCode ) {
+										event.preventDefault();
+										updateStyle( styleOption.value );
+									}
+								} }
+								role="button"
+								tabIndex="0"
+								aria-label={ styleOption.label }
+							>
+								<div className="block-editor-block-styles__item-preview is-opentable">
+									{ blockPreview( styleOption.value ) }
+								</div>
+								<div className="block-editor-block-styles__item-label">{ styleOption.label }</div>
+							</div>
+						);
+					} ) }
+				</div>
+			</PanelBody>
+			<PanelBody title={ __( 'Settings', 'jetpack' ) }>
+				<RestaurantPicker rids={ rid } onChange={ onPickerSubmit } />
+				<SelectControl
```

Comment:
Instead of offering that option, I wonder if it wouldn't be best to follow the site language settings; if the site language is a language supported by OpenTable use that, if not default to English?

Reply:
Good idea. Do you think we should have both? Default to the site language, but allow the user to override it?

Reply:
That would work too. I am not sure how many folks would need to override it. Maybe folks with multilanguage sites? I'll let you all decide what seems to be the best user experience here!

Reply:
Maybe we should punt this to V2, we dont know how popular this block will be yet...

Reply:
You may have seen that I've implemented this by having an inline script that sets the site locale in the `core/block-editor' store. I'm not sure if it's the most elegant of solutions, so opinions are welcome!